plugins/outputs/opentelemetry/opentelemetry_test.go

```diff
@@ -81,6 +86,190 @@ func TestOpenTelemetry(t *testing.T) {
 	require.JSONEq(t, string(expectJSON), string(gotJSON))
 }
 
+func TestOpenTelemetryHTTPProtobuf(t *testing.T) {
+	expect := pmetric.NewMetrics()
+	{
+		rm := expect.ResourceMetrics().AppendEmpty()
+		rm.Resource().Attributes().PutStr("host.name", "potato")
+		rm.Resource().Attributes().PutStr("attr-key", "attr-val")
+		ilm := rm.ScopeMetrics().AppendEmpty()
+		ilm.Scope().SetName("My Library Name")
+		m := ilm.Metrics().AppendEmpty()
+		m.SetName("cpu_temp")
+		m.SetEmptyGauge()
+		dp := m.Gauge().DataPoints().AppendEmpty()
+		dp.Attributes().PutStr("foo", "bar")
+		dp.SetTimestamp(pcommon.Timestamp(1622848686000000000))
+		dp.SetDoubleValue(87.332)
+	}
+
+	var receivedMetrics pmetric.Metrics
+	var receivedContentType string
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		receivedContentType = r.Header.Get("Content-Type")
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			t.Error(err)
+		}
+
+		req := pmetricotlp.NewExportRequest()
+		err = req.UnmarshalProto(body)
+		if err != nil {
+			t.Error(err)
+		}
```

Comment:
Please limit the scope of the variables! Same below.

```suggestion
		if err := req.UnmarshalProto(body); err != nil {
			t.Error(err)
			w.WriteHeader(http.StatusInternalServerError)
			return
		}
```

Reply:
Fixed. https://github.com/influxdata/telegraf/pull/17997/commits/52aa201df03125fe0d3aedcfd05a2ed24146ba44